File: FirstNotRepeatingChar/FirstNotRepeatingChar.py
# -*- coding:utf-8 -*-
class Solution:
    def FirstNotRepeatingChar(self, s):
        # write code here
        import collections
        n = len(s)
        res = -1
        if n==0:
            return res
        count = collections.Counter(s)
        for index,ch in enumerate(s):
            if count[ch]==1:
                return index
        return res

File: FirstNotRepeatingChar/FirstNotRepeatingChar.cpp
class Solution {
public:
    int FirstNotRepeatingChar(string s) {
        int res = -1;
        int n = s.size();
        if (n==0) return res;
        unordered_map<char,int> m;
        for (int i=0;i<n;i++)
        {
            m[s[i]]++;
        }
        for (int i=0;i<n;i++) 
        {
            if (m[s[i]]==1)
            {
                return i;
            }
        }
        return res;
    }
};
